src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -96,6 +108,18 @@ private static BicepCompletionContextKind GetDeclarationTypeFlags(IList<SyntaxBa
                     // the cursor position is on the type if we have an identifier (non-zero length span) and the offset matches the type position
                     return ConvertFlag(output.Name.Span.Length > 0 && output.Type.Span.Position == offset, BicepCompletionContextKind.OutputType);
 
+                case ResourceDeclarationSyntax resource:
+                    // the most specific matching node is a resource declaration
+                    // the declaration syntax is "resource <identifier> '<type>' ..."
+                    // the cursor position is on the type if we have an identifier (non-zero length span) and the offset matches the type position
+                    return ConvertFlag(resource.Name.Span.Length > 0 && resource.Type.Span.Position == offset, BicepCompletionContextKind.ResourceType);
+
+                case ModuleDeclarationSyntax module:
```

Comment:
The module-related changes are untested, but the logic is very similar to resources, so I just added it in. We'll consume it in another PR.